moveit_core/collision_detection/src/world.cpp

```diff
@@ -131,14 +131,63 @@ void World::ensureUnique(ObjectPtr& obj)
     obj.reset(new Object(*obj));
 }
 
-bool World::hasObject(const std::string& id) const
+bool World::hasObject(const std::string& object_id) const
 {
-  return objects_.find(id) != objects_.end();
+  return objects_.find(object_id) != objects_.end();
 }
 
-bool World::moveShapeInObject(const std::string& id, const shapes::ShapeConstPtr& shape, const Eigen::Isometry3d& pose)
+bool World::knowsTransform(const std::string& object_id) const
 {
-  auto it = objects_.find(id);
+  // Check object names first
```

Comment:
It looks like this code does what it's meant to, but i'm torn. It seems a bit sketchy to put both "objects" and "child objects" in the same object namespace. Based on the description of the PR, I was expecting these child frames to only make sense in the context of their parent objects. For example "grasp_pose" or "handle_pose" on a teapot aren't the name of an object at the top level. It feels like it would be cleaner to not treat these child frames "objects" at all; so it seems like a top level "knowsTransform" should only check for object transforms (and therefor wouldn't be too useful compared to  `hasObject()`.

Reply:
See the mainline comment